core/integration/features/settlement/0053-PERP-028_twap_calculation_with_auction.feature

```diff
@@ -0,0 +1,156 @@
+        When time is updated to "2020-10-16T00:16:00Z"
+        Then the oracles broadcast data with block time signed with "0xCAFECAFE1":
+            | name           | value | time offset |
+            | perp.ETH.value | 110   | -1s         |
+        And the trading mode should be "TRADING_MODE_MONITORING_AUCTION" for the market "ETH/DEC19"
+
+        # 7 mins in, the auction period will end
+        Given time is updated to "2020-10-16T00:17:01Z"
+        When the trading mode should be "TRADING_MODE_CONTINUOUS" for the market "ETH/DEC19"
+        Then the parties place the following orders:
+            | party  | market id | side | volume | price | resulting trades | type       | tif     |
+            | party1 | ETH/DEC19 | buy  | 10     | 9     | 0                | TYPE_LIMIT | TIF_GTC |
+            | party2 | ETH/DEC19 | sell | 10     | 9     | 1                | TYPE_LIMIT | TIF_GTC |
+
+        # 8 mins in, the auction period will end
+        Given time is updated to "2020-10-16T00:18:00Z"
+        When the trading mode should be "TRADING_MODE_CONTINUOUS" for the market "ETH/DEC19"
+        And the markets are updated:
+            | id        | price monitoring | linear slippage factor | quadratic slippage factor |
+            | ETH/DEC19 | default-none     | 1e6                    | 1e6                       |
+        Then the parties place the following orders:
+            | party  | market id | side | volume | price | resulting trades | type       | tif     |
+            | party1 | ETH/DEC19 | buy  | 1      | 8     | 0                | TYPE_LIMIT | TIF_GTC |
+            | party2 | ETH/DEC19 | sell | 1      | 8     | 1                | TYPE_LIMIT | TIF_GTC |
+        And the oracles broadcast data with block time signed with "0xCAFECAFE1":
+            | name           | value | time offset |
+            | perp.ETH.value | 80    | -1s         |
+
+        # 9 mins in, the auction period will end
+        Given time is updated to "2020-10-16T00:19:00Z"
+        Then the oracles broadcast data with block time signed with "0xCAFECAFE1":
+            | name           | value | time offset |
+            | perp.ETH.value | 140   | -1s         |
+
+        # 10 mins in, the auction period will end
+        Given time is updated to "2020-10-16T00:20:00Z"
+        When the parties place the following orders:
+            | party  | market id | side | volume | price | resulting trades | type       | tif     |
+            | party1 | ETH/DEC19 | buy  | 1      | 30    | 0                | TYPE_LIMIT | TIF_GTC |
+            | party2 | ETH/DEC19 | sell | 1      | 30    | 1                | TYPE_LIMIT | TIF_GTC |
+        # in theory internal TWAP = 9.625 external TWAP = 10.25, if the auction period is excluded
+        # but these are type int so the decimal is truncated
+        Then the product data for the market "ETH/DEC19" should be:
+            | internal twap | external twap |
+            | 9             | 10            |
```

Comment:
Given that these numbers end up being the same as the test where there isn't any auction periods, I think we need to scale up the prices/set-different-decimal-places-in-the-markets so that the TWAP calculations do not lose all the precision.

Reply:
change the mark/spot price during auction so that if the auction was included in the calculation then the twap values are off by a whole number